### PR TITLE
feat: switch projections & allow advanced projections

### PR DESCRIPTION
**Pull request by RobertOrthofer**
This PR brings the usage of projections, as described in #702.

Out-of-the-box, `EPSG:4326` is supported in addition to the standard `EPSG:3857`. Other projections can be used by calling the helper-function `registerProjection` first, including a name and proj4 definition string. These can be obtained by via services like [epsg.io](epsg.io), this must be handled by the parent application.

Switching the projection is possible by setting the `projection`-attribute. A new `view` is created when doing so, listener functions are moved over to the new view. It is important that set listeners retrieve the view via `evt.target`, instead of `this`.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added a test related to this feature/fix
- [x] For new features: I have created a story showcasing this new feature
- [x] All checks have passed
- [x] The PR title [follows conventional commit style and reflects the type of change (major/minor/patch, breaking)](https://github.com/googleapis/release-please?tab=readme-ov-file#how-should-i-write-my-commits)
- [x] The PR title describes the change within this element (each merged PR equals one entry in the element's changelog!)

### CODE REVIEW

File: elements/map/helpers.ts [hunk 1]
```diff
@@ -213,3 +215,11 @@ export function isTopoJSON(text: string): boolean {
     return false;
   }
 }
+
+export function registerProjection(
+  name: string,
+  projection: string | proj4.ProjectionDefinition
```
Comment (silvester-pari) [inline elements/map/helpers.ts:221]: Wouldn't it be a very nice convenience function to make this async and fetch the definition for the user from https://epsg.io/?format=json&q=? Like in https://openlayers.org/en/latest/examples/reprojection-by-code.html.

File: elements/map/main.ts [hunk 2]
```diff
@@ -208,6 +212,57 @@ export class EOxMap extends LitElement {
...
+  }
+
+  /**
+   * projection of the map view as SRS-identifier (e.g. EPSG:4326)
+   */
+  @property({ attribute: "projection", type: String })
+  set projection(projection: ProjectionLike) {
+    const oldView = this.map.getView();
+    if (projection && projection !== oldView.getProjection().getCode()) {
+      const newCenter = transform(
```
Comment (silvester-pari) [inline elements/map/main.ts:231]: I didn't manage to successfully test this. When changing the projection from e.g. 3857 to 4326, then the zoom is reset to initial. I tried in storybook:

```
      <button
        @click=${() => {
          const eoxMap = document.querySelector("eox-map");
          eoxMap.setAttribute("projection", "EPSG:4326");
        }}
      >
        change
      </button>
```

Initial center: `center: [16.8, 48.2],`

**Review by silvester-pari — CHANGES_REQUESTED** [review]
Looks good in general, thank you! Please see the two comments, and please add a story showcasing this new functionality :slightly_smiling_face:

**Review by silvester-pari — APPROVED** [review]

File: elements/map/helpers.ts (hunk 1 above)
Comment (RobertOrthofer) [reply elements/map/helpers.ts:237]: i added this as another helper functionality, utilizing `fromEPSGCode`

File: elements/map/main.ts (hunk 2 above)
Comment (RobertOrthofer) [reply elements/map/main.ts:258]: i couldn't reproduce this behaviour with zoom, but i did find weird interactions with the initial center, which came down with weird interactions and timings when setting the center of the view and setting the target of the map... this was quite tough to find, i restructured the `center` to be in line with the other attributes (with a getter and setter instead of watching for changes in the `updated`-hook), and setting the center once after the map was added to the `DOM`.

Also added an entry to the story book

**Comment by lubojr** [comment]
@RobertOrthofer Thank you very much for this new feature! :rocket: 
I have an implementation detail question - what will happen now if the old "center" of current view can not be projected into the new projection? Will it not throw an openlayers exception?

Example: When using a polar stereographic projection centered on pole and decide to switch my map to EPSG:3857 which has projection bounds geographically around 85 degrees so my original center will be unprojectable. Would it be a good idea to in that case set center to new projection center ?

**Comment by RobertOrthofer** [comment]
that is a great question, do you have an EPSG-code to test this out? I have intentionally left out setting the `extent` of the view to the `extent` of the projection for now, as this might limit some very edge cases

**Comment by lubojr** [comment]
@RobertOrthofer Let's try this out. 

For example the North Polar stereographic can be used: 

proj4 def:
```
{
  name: 'EPSG:3411',
  def: '+proj=stere +lat_0=90 +lat_ts=70 +lon_0=-45 +x_0=0 +y_0=0 +a=6378273 +b=6356889.449 +units=m +no_defs +type=crs',
  extent: [-3314763.31, -3314763.31, 3314763.31, 3314763.31],
}
```

**Comment by RobertOrthofer** [comment]
I tried this with the `registerProjectionFromCode`-helper and it looks like it always manages to reproject the center. Obviously, if the center is way off the projections bounds, the map becomes quite distorted, but usable...
I think for now this would be the responsibility of the application to move the map into the area of interest beforehand, as limiting the extent might not always be the desired behaviour

**Comment by lubojr** [comment]
Perfect! Thanks for checking!

**Review by lubojr — APPROVED** [review]

**Review by silvester-pari — APPROVED** [review]
